File: abstractfactory/java/IosControllerFactory.java
package io.github.androidhot.designpattern.abstractfactory.java;

public class IosControllerFactory implements ControllerFactory {
    @Override public InterfaceController createInterfaceController() {
        return new IosInterfaceController();
    }


    @Override public OperationController createOperationController() {
        return new IosOperationController();
    }
}

File: abstractfactory/kotlin/IosControllerFactory.kt
package io.github.androidhot.designpattern.abstractfactory.kotlin

class IosControllerFactory : ControllerFactory {
    override fun createInterfaceController(): InterfaceController {
        return IosInterfaceController()
    }

    override fun createOperationController(): OperationController {
        return IosOperationController()
    }
}
